### ISSUE TITLE
Dalaload processing

### ISSUE THREAD

**Issue by chamecall** [adamzjk/Age-Gender-Pred#10]
Can you pls explain what does this sample of code:

`age_cls_unit = int(parser['RacNet']['age_cls_unit'])

# distribution of IMDB-WIKi dataset I: IMDB-Wiki
imdb_distr = [25, 63, 145, 54, 46, 113, 168, 232, 455, 556,
                752, 1089, 1285, 1654, 1819, 1844, 2334, 2828,
                3346, 4493, 6279, 7414, 7706, 9300, 9512, 11489,
                10481, 12483, 11280, 13096, 12766, 14346, 13296,
                12525, 12568, 12278, 12694, 11115, 12089, 11994,
                9960, 9599, 9609, 8967, 7940, 8267, 7733, 6292,
                6235, 5596, 5129, 4864, 4466, 4278, 3515, 3644,
                3032, 2841, 2917, 2755, 2853, 2380, 2169, 2084,
                1763, 1671, 1556, 1343, 1320, 1121, 1196, 949,
                912, 710, 633, 581, 678, 532, 491, 428, 367,
                339, 298, 203, 258, 161, 136, 134, 121, 63, 63,
                82, 40, 37, 24, 16, 18, 11, 4, 9]
imdb_distr[age_cls_unit - 1] = sum(imdb_distr[age_cls_unit - 1:])
imdb_distr = imdb_distr[:age_cls_unit]
imdb_distr = np.array(imdb_distr, dtype='float')

# distribution of test dataset: FG-NET
fg_distr = [10, 10, 10, 10, 10, 10, 10, 10, 10, 9, 10, 10, 10, 10, 10, 10, 10, 10, 10,
            9, 8, 8, 9, 9, 5, 7, 6, 6, 7, 6, 9, 5, 4, 6, 5, 7, 6, 3, 3, 5, 5, 4, 4, 2,
            3, 5, 2, 2, 2, 3, 2, 3, 3, 2, 2, 2, 0, 0, 1, 0, 1, 3, 1, 1, 0, 0, 0, 1, 0, 0]
fg_distr[age_cls_unit - 1] = sum(fg_distr[age_cls_unit - 1:])
fg_distr = fg_distr[:age_cls_unit]
fg_distr = np.array(fg_distr, dtype='float') + 1

# step 1: correct different distribution between datasets
loss_weight = fg_distr / imdb_distr

# step 2: normalize the weight so that the expected weight for a random sample
#         from training dataset equals to 1, i.e. sum(weight * 1/imdb_distr ) = 1
loss_weight = loss_weight / sum(loss_weight / imdb_distr)

# >>> (loss_weight * 100).astype('int')
# array([1398,  554,  241,  647,  760,  309,  208,  150,   76,   57,   46,
#          32,   27,   21,   19,   18,   14,   12,   10,    7,    4,    3,
#           4,    3,    2,    2,    2,    1,    2,    1,    2,    1,    1,
#           1,    1,    2,    1,    1,    1,    1,    1,    1,    1,    1,
#           1,    2,    1,    1,    1,    2,    1,    2,    2,    2,    2,
#           2,    1,    1,    2,    0])

loss_weight = torch.from_numpy(np.array(loss_weight, dtype='float'))
loss_weight = loss_weight.type(torch.FloatTensor)`
